File: python/pysearch/fileutil.py
# -*- coding: utf-8 -*-
################################################################################
#
# fileutil.py
#
# class FileUtil: provides utility functions for getting file extension and
#                 determining file type
#
################################################################################
import os

class FileUtil(object):
    """a file helper class"""

    DOT_DIRS = ('.', '..')

    @staticmethod
    def get_extension(filename):
        """Returns the extension for a given filename, if any, else empty 
           string"""
        ext = ''
        if os.path.basename(filename).rfind('.') > 0:
            ext = filename.split('.')[-1]
        return ext.lower()

    @staticmethod
    def is_hidden(filename):
        """Returns true if file is hidden else false"""
        f = os.path.basename(filename)
        if len(f) > 1 and f.startswith('.') and not f in FileUtil.DOT_DIRS:
            return True
        return False

File: ruby/rbsearch/fileutil.rb
# frozen_string_literal: true

# FileUtil - file utility functions
module FileUtil
  module_function

  def dot_dirs
    %w[. .. ./ ../]
  end

  def dot_dir?(filename)
    f = File.basename(filename)
    dot_dirs.include?(f)
  end

  def get_extension(filename)
    ext = ''
    f = File.basename(filename)
    index = f.rindex('.')
    if index&.positive? && index < (f.length - 1)
      ext = f[index + 1..f.length].downcase
    end
    ext
  end

  def hidden?(filename)
    f = File.basename(filename)
    if f.length > 1 && f[0] == '.' && !dot_dirs.include?(f)
      true
    else
      false
    end
  end
end
